src/category_theory/limits/preserves/shapes/images.lean

```diff
@@ -0,0 +1,72 @@
+/-
+Copyright (c) 2022 Jujian Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jujian Zhang
+-/
+import category_theory.limits.shapes.images
+import category_theory.limits.constructions.epi_mono
+
+/-!
+# Preserving images
+
+In this file, we show that if a functor preserves span and cospan, then it preserves images.
+-/
+
+
+noncomputable theory
+
+namespace category_theory
+
+namespace preserves_image
+
+open category_theory
+open category_theory.limits
+
+universes u₁ u₂ v₁ v₂
+
+variables {A : Type u₁} {B : Type u₂} [category.{v₁} A] [category.{v₂} B]
+variables [has_equalizers A] [has_images A]
+variables [strong_epi_category B] [has_images B]
+variables (L : A ⥤ B)
+variables [Π {X Y Z : A} (f : X ⟶ Z) (g : Y ⟶ Z), preserves_limit (cospan f g) L]
+variables [Π {X Y Z : A} (f : X ⟶ Y) (g : X ⟶ Z), preserves_colimit (span f g) L]
+
+/--
+If a functor preserves span and cospan, then it preserves images.
+-/
+def iso {X Y : A} (f : X ⟶ Y) : image (L.map f) ≅ L.obj (image f) :=
```

Comment:
Suggestion: you add `@[simps]` here. You also add the lemma
```lean
@[simp]
lemma is_image_lift {X Y : B} (f : X ⟶ Y) (F : mono_factorisation f) :
  (image.is_image f).lift F = image.lift F :=
rfl
```
to `limits/shapes/images.lean` (after the definition of `image.lift` around line 285).

Then the three lemmas below are all proved `by simp`. Please add  `@[reassoc]` to them.